core/tee/tee_svc_storage.c

```diff
@@ -710,7 +710,13 @@ TEE_Result syscall_storage_start_enum(unsigned long obj_enum,
 		return TEE_ERROR_ITEM_NOT_FOUND;
 
 	e->fops = fops;
-	assert(!e->dir);
+
+	if (e->fops && e->dir) {
+		e->fops->closedir(e->dir);
```

Comment:
What if `e->dir` was previously opened with a different `fops`? Not sure it can actually happen but it might be safer to close before doing `e->fops = fops;` (right after `tee_svc_storage_get_enum()`).

Reply:
I'm not sure, but I think that the "dir-handle" that we get is the one that we shall close. Does it matter where that is done? Having that said, I can move it up if we believe that makes sense.